whelk-core/src/main/groovy/whelk/converter/marc/MarcFrameConverter.groovy

```diff
@@ -953,7 +954,40 @@ class ConversionPart {
             aboutMap = (Map<String, List>) state.aboutMap
         }
 
-        return (Map) aboutMap[aboutEntityName]?.get(0) ?: data
+        Map aboutEntity = (Map) aboutMap[aboutEntityName]?.get(0) ?: data
+        if (fallbackEntityName) {
+            Map fallbackEntity = (Map) aboutMap[fallbackEntityName]?.get(0)
+            if (fallbackEntity)
+                return deepMergedClone(fallbackEntity, aboutEntity)
+        }
+        return aboutEntity
+    }
+
+    Map deepMergedClone(Map keepSome, Map keepAll) {
+        def result = [:]
+        Set keySet = keepAll.keySet() + keepSome.keySet()
+        for (Object key : keySet) {
+
+            Object highPrioValue = keepAll[key]
+            Object lowPrioValue = keepSome[key]
+
+            if (!highPrioValue && lowPrioValue)
+                result.put(key, lowPrioValue)
+            else if (highPrioValue && !lowPrioValue)
+                result.put(key, highPrioValue)
+            else if (highPrioValue && lowPrioValue) {
+                if (highPrioValue instanceof Map && lowPrioValue instanceof Map) {
+                    result.put(key, deepMergedClone((Map)lowPrioValue, (Map)highPrioValue))
+                }
+                else if (highPrioValue instanceof List && lowPrioValue instanceof List) {
+                    List resultingList = (List) highPrioValue
+                    resultingList.addAll( (List) lowPrioValue )
```

Comment:
This mutates the input. Change this to:
```suggestion
                    List resultingList = []
                    resultingList.addAll( highPrioValue )
                    resultingList.addAll( lowPrioValue )
```
and all tests pass. (No need to cast thanks to the `instanceof` test.)

Reply:
Nice catch! Thanks! :)

I had been thinking all through easter, that id mutated the input, but didnt have time to check it. You beat me to it! Thanks!